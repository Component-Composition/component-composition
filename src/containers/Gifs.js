import React, { Component } from 'react';
import Cards from '../components/DeckOfCards';

export default class Gifs extends Component {
  state = {
    data: []
  }

  componentDidMount() {
    fetch('https://hey-arnold-api.herokuapp.com/api/v1/gifs')
      .then(response => response.json())
      .then(data => this.setState({ data }));
  }

  render() {
    const { data } = this.state;
    return (
      <Cards characters={data} />
    );

  }

}
