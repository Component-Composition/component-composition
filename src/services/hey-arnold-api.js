function getItems(url) {
  return fetch(url)
    .then(response => response.json());
}

export { getItems };
