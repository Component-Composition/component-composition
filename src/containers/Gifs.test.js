import React from 'react';
import { shallow } from 'enzyme';
import Gifs from './Gifs';
import { getItems } from '../services/hey-arnold-api';

jest.mock('../services/hey-arnold-api.js', () => {
  return {
    getItems() {
      return Promise.resolve([{
        '_id': '5da237699734fdcb7bef8f51',
        'gifLink': '"https://media.tenor.com/images/e7623921965651fa6fba7e0c37892044/tenor.gif"'
      }]);
    }
  };
});

describe('Gifs Component', () => {
  it('renders Characters', () => {
    return getItems().then(item => {
      const wrapper = shallow(<Gifs items={item} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
