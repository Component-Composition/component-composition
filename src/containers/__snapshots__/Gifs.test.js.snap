// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifs Component renders Characters 1`] = `
<Cards
  items={Array []}
/>
`;
